lib/logstash/outputs/elasticsearch.rb

```diff
@@ -526,9 +526,11 @@ def receive(event)
       :_routing => @routing ? event.sprintf(@routing) : nil
     }
     
-    params[:_upsert] = LogStash::Json.load(event.sprintf(@upsert)) if @action == 'update' && @upsert != ""
+    action = event.sprintf(@action)
```

Comment:
have you considered naming this something more descriptive? Say, `sprinted_action`?